c7n/resources/aws.py

```diff
@@ -614,19 +614,19 @@ def initialize_policies(self, policy_collection, options):
                 _, resource_type = p.resource_type.split('.', 1)
             else:
                 resource_type = p.resource_type
-            available_regions = service_region_map.get(
-                resource_service_map.get(resource_type), ())
+            available_regions = set(service_region_map.get(
+                resource_service_map.get(resource_type), ()))
 
             # its a global service/endpoint, use user provided region
             # or us-east-1.
             if not available_regions and options.regions:
-                candidates = [r for r in options.regions if r != 'all']
-                candidate = candidates and candidates[0] or 'us-east-1'
-                svc_regions = [candidate]
+                svc_regions = {next(
+                    (r for r in options.regions if r != 'all'), 'us-east-1')}
             elif 'all' in options.regions:
-                svc_regions = list(set(available_regions).intersection(enabled_regions))
+                svc_regions = available_regions.intersection(enabled_regions)
             else:
-                svc_regions = options.regions
+                svc_regions = set(options.regions)
```

Comment:
we should ideally be preserving order for region processing from what was passed in on the cli.